Cookbook/Proposals/Base class for FeedbackLoops systems.md

```diff
@@ -0,0 +1,157 @@
+# Base class for FeedbackLoops systems
+
+* Author(s): Sergey Shulga
+* Review Manager: Anders Ha
+
+## Introduction
+
+Almost for all our screens, we implement ViewModels which are powered by the [ReactiveFeeedback](https://github.com/Babylonpartners/ReactiveFeedback).
+
+The current interface of VM looks like this
+
+```swift
+public protocol BoxViewModel: AnyObject {
+    associatedtype State
+    associatedtype Action
+
+    var state: Property<State> { get }
+
+    func send(action: Action)
+}
+```
+
+## Motivation
+
+One of the main issues with this approach
+
+1. We need to implement `input` feedback in pretty much every VM which may lead to errors if do not schedule input correctly, for this we have `Feedback.input()` convenience extension
+
+2. If the screen contains a lot of input field we end up having a lot of event's that are just mutate one property of the `State` e.g
+`Action.didChangeUserName(String)` `Action.didPickBirthDate(Date)`
+ 
+## Proposed solution
+1. Implementation of the feedback:
+
+```swift
+open class FeedbackLoop<State, Event, Action>: FeedbackLoopType {
+  public let state: Property<State>
+  private let input = Feedback<State, Update>.input()
+  private let actionTransformer: (Action) -> Event
+
+  public init(
+    initial: State,
+    reducer: @escaping (State, Event) -> State,
+    feedbacks: [Feedback<State, Event>],
+    scheduler: DateScheduler,
+    actionTransformer: @escaping (Action) -> Event
+  ) {
+    self.actionTransformer = actionTransformer
+    self.state = Property(
+      initial: initial,
+      scheduler: scheduler,
+      reduce: { (state: State, update: Update) -> State in
+        switch update {
+        case let .event(event):
+          return reducer(state, event)
+        case let .mutation(mutation):
+          return mutation.mutate(state)
+        }
+      },
+      feedbacks: feedbacks.map { $0.mapEvent(Update.event) }
```

Comment:
what is this `mapEvent`?

Reply:
See here: https://github.com/sergdort/CombineFeedback/blob/master/Sources/CombineFeedbackUI/ViewModel.swift#L62-L66

Reply:
I guess in terms of point free it's a form of pull back 🤔

Reply:
A form of pull back would be on the state (versus the event):

```
func mapEvent<U>(_ f: (Event -> U) -> Feedback<State, U> {

}
```

versus

```
func contraMapEvent<U>(_ f: (State -> U) -> Feedback<U, Event> {

}
```